File: Domain/Floorplan.cs
using System;
using System.Collections.Generic;

namespace Domain;

public class Floorplan
{
    public Guid Id {get; set;}
    public string Name {get; set;}
    
    public int FloorNumber { get; set; }
    
    public string Image {get; set;}
    public string Scale {get; set;}
    
   // public IEnumerable<Point> Nodes {get; set;}
   public HashSet<Point> Points {get; set;} // Huge performance improvement allemaal zelfde oplook tijd 
    public Hospital Hospital {get; set;}
    
    public Floorplan(string name, int floorNumber, string scale, string image)
    {
        Name = name;
        FloorNumber = floorNumber;
        Scale = scale;
        Image = image;
    }
}

File: ui/lib/Models/Floorplan.dart
import 'Point.dart';
import 'Hospital.dart';


class Floorplan {
  final String id;
  final String name;
  final int floorNumber;
  final String scale;
  final List<Point> nodes;
  final Hospital hospital;
  final String imageData;

  Floorplan({
    required this.id,
    required this.name,
    required this.floorNumber,
    required this.scale,
    required this.nodes,
    required this.hospital,
    required this.imageData,
  });

  factory Floorplan.fromJson(Map<String, dynamic> json) {
    return Floorplan(
      id: json['id'],
      name: json['name'],
      floorNumber: json['floorNumber'],
      scale: json['scale'],
      nodes: (json['nodes'] as List).map((i) => Point.fromJson(i)).toList(),
      hospital: Hospital.fromJson(json['hospital']),
      imageData: json['imageData'],
    );
  }
}